packages/qdrant-loader/src/qdrant_loader/core/pipeline/orchestrator.py

```
@@ -79,11 +88,50 @@ async def _process_source_type(source_type_name: str, source_configs):
             if not source_configs:
                 return
 
+            # Tracks the checkpoint cursor of the last document with checkpoint
+            # info, so a batch is flushed before crossing a page boundary
+            # (see _process_source_type body below for why this matters).
+            last_cursor_value = None
+            # True once a size-based mid-page flush has fired for the current page.
+            # Used to emit the warning only once per overflowing page.
+            page_has_overflowed = False
+
+            async def connector_factory_with_checkpoint(src_config):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `connector_factory_with_checkpoint`

(ANN202)